packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -868,23 +842,6 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     }
 
     public get storage(): IDocumentStorageService {
-        // This code is plain wrong. It lies that it never returns undefined!!!
-        // All callers should be fixed, as this API is called in detached state of container when we have
-        // no storage and it's passed down the stack without right typing.
-        // back-compat 0.40 NoStorageInDetachedMode. Also, IContainerContext storage api return type still
-        // has undefined in 0.39 package version.
-        // So once we release 0.40 container-defn package we can remove this check.
-        if (!this._storage && this.context.storage) {
-            // Note: BlobAggregationStorage is smart enough for double-wrapping to be no-op
-            // If isolated channels are disabled, then there are no .channel layers, we pack at level 1,
-            // otherwise we pack at level 2
-            this._storage = BlobAggregationStorage.wrap(
```

Comment:
I do not know if this code is right or wrong, but it's worth pointing out that it would not return undefined if we have this.context.storage. I.e. this.context.storage !== undefined && this._storage === undefined => this._storage is initialized and returned.
With this change we will return this._storage as is, and if it was undefined, we will return undefined.
Maybe it's correct, but it's definitely change in behavior that we need to look closer and confirm it's correct.

Reply:
Looking at the code, `this.context.storage` is always defined. Also, the code before blob aggregation was added returned simply returned `this.context.storage` here. We should do just that now.

Reply:
yes, we can easily remove it - this.context.storage  === this._storage.

Reply:
Ok, thanks for validating!